blockchain/db/postgres/transaction_db.py

```diff
@@ -218,6 +218,55 @@ def get_all(limit=None, offset=None, **params):
         get_connection_pool().putconn(conn)
 
 
+def get_subscription_txns(criteria, block_id=None):
+    """ retrieve transactions that meet given criteria and have a block_id >= minimum_block_id """
+    query = SQL_GET_ALL
+    query += """ WHERE """
+    separator_needed = False
+
+    if "transaction_type" in criteria:
+        query += """ transaction_type = '""" + str(criteria["transaction_type"]) + """'"""
```

Comment:
It looks like psycopg supports named arguments (`%(name)s`) within a query, eg:
```
cur.execute(
...     """INSERT INTO some_table (an_int, a_date, another_date, a_string)
...         VALUES (%(int)s, %(date)s, %(date)s, %(str)s);""",
...     {'int': 10, 'str': "O'Reilly", 'date': datetime.date(2005, 11, 18)})
```

This would still provide flexibility around building queries without worrying about how many or the order of the `%s` params.